fe/fe-core/src/main/java/org/apache/doris/planner/external/iceberg/IcebergMetadataCache.java

```diff
@@ -0,0 +1,265 @@
+                .forEach(entry -> {
+                    ManifestFiles.dropCache(entry.getValue().io());
+                    tableCache.invalidate(entry.getKey());
+                });
+    }
+
+    public void invalidateTableCache(long catalogId, String dbName, String tblName) {
+        snapshotListCache.asMap().keySet().stream()
+            .filter(key -> key.catalogId == catalogId && key.dbName.equals(dbName) && key.tableName.equals(tblName))
+            .forEach(snapshotListCache::invalidate);
+
+        tableCache.asMap().entrySet().stream()
+                .filter(entry -> {
+                    IcebergMetadataCacheKey key = entry.getKey();
+                    return key.catalogId == catalogId && key.dbName.equals(dbName) && key.tableName.equals(tblName);
+                })
+                .forEach(entry -> {
+                    ManifestFiles.dropCache(entry.getValue().io());
+                    tableCache.invalidate(entry.getKey());
+                });
+    }
+
+    public void invalidateDbCache(long catalogId, String dbName) {
+        snapshotListCache.asMap().keySet().stream()
+                .filter(key -> key.catalogId == catalogId && key.dbName.equals(dbName))
+                .forEach(snapshotListCache::invalidate);
+
+        tableCache.asMap().entrySet().stream()
+                .filter(entry -> {
+                    IcebergMetadataCacheKey key = entry.getKey();
+                    return key.catalogId == catalogId && key.dbName.equals(dbName);
+                })
+                .forEach(entry -> {
+                    ManifestFiles.dropCache(entry.getValue().io());
+                    tableCache.invalidate(entry.getKey());
+                });
+    }
+
+    private Table createIcebergTable(String uri, Map<String, String> hdfsConf, String db, String tbl) {
+        // set hdfs configure
+        Configuration conf = new HdfsConfiguration();
+        for (Map.Entry<String, String> entry : hdfsConf.entrySet()) {
+            conf.set(entry.getKey(), entry.getValue());
+        }
+
+        HiveCatalog hiveCatalog = new HiveCatalog();
+        hiveCatalog.setConf(conf);
+
+        Map<String, String> catalogProperties = new HashMap<>();
+        catalogProperties.put(HMSProperties.HIVE_METASTORE_URIS, uri);
+        catalogProperties.put("uri", uri);
+        hiveCatalog.initialize("hive", catalogProperties);
+
+        Table table = hiveCatalog.loadTable(TableIdentifier.of(db, tbl));
+
+        initIcebergTableFileIO(table);
+
+        return table;
+    }
+
+    private Table createIcebergTable(HMSExternalTable hmsTable) {
+        return createIcebergTable(hmsTable.getMetastoreUri(),
+            hmsTable.getHadoopProperties(),
+            hmsTable.getDbName(),
+            hmsTable.getName());
+    }
+
+    private void initIcebergTableFileIO(Table table) {
+        Map<String, String> ioConf = new HashMap<>();
+        table.properties().forEach((key, value) -> {
```

Comment:
Is table.properties() and CatalogProperty different? Is it missing some properties such as s3 or hdfs HA?